### PR TITLE
feat(project): add TypeScript definitions

### PR DESCRIPTION
**Pull request by nikeee**
Are `t1`, `t2`, `bez1` and `bez2` part of the public API? They aren't tested for in the tests. However, I think diagram-js uses some of it somewhere.

### CODE REVIEW

**Comment by nikku** [comment]
Yes, these values are needed to properly fix / sort intersections in certain edge cases. Cf. [diagram-js usage](https://github.com/bpmn-io/diagram-js/blob/645177fa20b0320e702fceccee7a62e5f044e008/lib/layout/LayoutUtil.js).

We should figure out what they are doing and document them.

**Comment by nikku** [comment]
I'll have a look and provide you with some pointers.

**Comment by nikku** [comment]
Documentation:

`bez1`: bezier curve for matching path segment 1
`bez2`: bezier curve for matching path segment 2
`t1`: relative position of intersection on path segment1 (0.5 => in middle, 0.0 => at start, 1.0 => at end)
`t2`: relative position of intersection on path segment2 (...)

While you're on it, you should probably document `segment1`, `segment2`, `x`, `y` on `Intersection`, too.

I'll add a test that covers the additional data exposed by `path-intersection` as well as the counting use-case.

**Comment by nikeee** [comment]
Are these properties set in every case? Or just in some cases?

**Comment by nikku** [comment]
These properties are set in every case (excluding counting of course).

**Comment by nikku** [comment]
I've added additional test coverage via https://github.com/bpmn-io/path-intersection/commit/003089780039fabcfad806f06c3546feeb509301 and [documented the main library export](https://github.com/bpmn-io/path-intersection/blob/master/intersect.js#L414). Please rebase your PR on top of these changes.

**Comment by nikeee** [comment]
Done

File: intersect.d.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * var intersections = findPathIntersections(
+ *   'M0,0L100,100',
+ *   [ [ 'M', 0, 100 ], [ 'L', 100, 0 ] ]
+ * );
+ *
+ * // intersections = [
+ * //   { x: 50, y: 50, segment1: 1, segment2: 1, t1: 0.5, t2: 0.5 }
+ * //
+ *
+ * @param {String|Array<PathDef>} path1
```
Comment (nikku) [inline intersect.d.ts]: Shouldn't `PathDef` be `PathComponent` (the exported type)?

Comment (nikeee) [reply intersect.d.ts]: Yes. Concerting TypeScript: If explicit type annotations are set, the type hints in the corresponding JSDoc are ignored. It wouldn't be an error, but it would be inconsistent with the docs.

I fixed this. Sorry!

**Comment by nikku** [comment]
Released as `v1.1.0`. Thanks!
